benchmarks/benchmark_functions/pl-braket-pipeline.py

```diff
@@ -0,0 +1,223 @@
+	device = qml.device("braket.aws.qubit",
+						device_arn='arn:aws:braket:::device/quantum-simulator/amazon/tn1',
+						s3_destination_folder=s3,
+						wires=n_wires,
+						shots=shots)
+
+	@qml.qnode(device, interface=interface, diff_method=diff_method)
+	def circuit(params_):
+		qml.templates.BasicEntanglerLayers(params_, wires=range(n_wires))
+		return qml.expval(qml.PauliZ(0))
+
+	for _ in range(num_repeats):
+
+		params = pnp.array(params, requires_grad=True)
+		opt = qml.GradientDescentOptimizer(stepsize=0.1)
+
+		for i in range(n_steps):
+			params = opt.step(circuit, params)
+
+
+def benchmark_power_tn1(bucket, prefix):
+	"""
+	Performs a QAOA optimization on 20 qubits with 5 layers on the remote Braket TN1 simulator.
+	"""
+
+	n_layers = 5
+	graph = nx.complete_graph(20)
+	params = [[0.5] * n_layers, [0.5] * n_layers]
+	interface = 'autograd'
+	diff_method = 'best'
+	n_wires = len(graph.nodes)
+	H_cost, H_mixer = qaoa.min_vertex_cover(graph, constrained=False)
+	shots = 1000
+
+	s3 = (bucket, prefix)
+	device = qml.device("braket.aws.qubit",
+						device_arn='arn:aws:braket:::device/quantum-simulator/amazon/tn1',
+						s3_destination_folder=s3,
+						wires=n_wires,
+						shots=shots)
+
+	def qaoa_layer(gamma, alpha):
+		qaoa.cost_layer(gamma, H_cost)
+		qaoa.mixer_layer(alpha, H_mixer)
+
+	@qml.qnode(device, interface=interface, diff_method=diff_method)
+	def circuit(params):
+		for w in range(n_wires):
+			qml.Hadamard(wires=w)
+		qml.layer(qaoa_layer, n_layers, params[0], params[1])
+		return [qml.sample(qml.PauliZ(i)) for i in range(n_wires)]
+
+	circuit(params)
+
+
+def benchmark_casual_ionq(bucket, prefix):
+	"""Trains a small quantum circuit for 20 steps using the remote Braket IonQ hardware.
+	The entire training is repeated 5 times.
+	"""
+	num_repeats = 5
+	n_steps = 20
+	n_wires = 4
+	n_layers = 6
+	interface = 'autograd'
+	params = random(size=(n_layers, n_wires))
+	diff_method = 'best'
+	shots = 1000
+
+	s3 = (bucket, prefix)
+	device = qml.device("braket.aws.qubit",
+						device_arn='arn:aws:braket:::device/qpu/ionq/ionQdevice',
+						s3_destination_folder=s3,
+						wires=n_wires,
+						shots=shots)
+
+	@qml.qnode(device, interface=interface, diff_method=diff_method)
+	def circuit(params_):
+		qml.templates.BasicEntanglerLayers(params_, wires=range(n_wires))
+		return qml.expval(qml.PauliZ(0))
+
+	for _ in range(num_repeats):
+
+		params = pnp.array(params, requires_grad=True)
+		opt = qml.GradientDescentOptimizer(stepsize=0.1)
+
+		for i in range(n_steps):
+			params = opt.step(circuit, params)
+
+
+def benchmark_power_ionq(bucket, prefix):
+	"""
+	Performs a QAOA optimization on 20 qubits with 5 layers on the remote Braket IonQ hardware.
+	"""
+
+	n_layers = 5
+	graph = nx.complete_graph(20)
+	params = [[0.5] * n_layers, [0.5] * n_layers]
+	interface = 'autograd'
+	diff_method = 'best'
+	n_wires = len(graph.nodes)
+	H_cost, H_mixer = qaoa.min_vertex_cover(graph, constrained=False)
+	shots = 1000
```

Comment:
This is a number to be mindful of (since one pays per shot, and because IonQ execution times are longer)